### PR TITLE
fix: prometheus with multiple providers

### PR DESCRIPTION
**Pull request by iwpnd**
we encountered an issue where tegola would panic if you'd try to register duplicate metrics collectors.
that would happen if you have more than one mvt provider in the same config file.

this fix makes provider metrics unique by adding the respective provider name to an array of constant labels. Upon collecting metrics on start up we'd pass unique metrics to the registration step and avoid said panic.

resolves: #886

### CODE REVIEW

File: internal/build/build.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 var (
-	Version              = "Version not set"
+	Version              = "version not set"
 	GitRevision          = "not set"
 	GitBranch            = "not set"
 	uiVersionDefaultText = "Viewer not build"
```
Comment (gdey) [inline internal/build/build.go:14]: Since we are fixing the Casing, could we fix it here as well. Thank you. 
"Viewer not built" -> "viewer not built"

**Review by gdey — APPROVED** [review]
LGTM

This is so much simpler, and the better solution! Thank you!

File: internal/build/build.go (hunk 1 above)
Comment (iwpnd) [reply internal/build/build.go:14]: missed that one, my bad.

**Comment by iwpnd** [comment]
@ARolek @gdey 

Can I suggest we merge #916 to fix the viewer in CI, then #893 to add the HANA provider and then #915 after I added unique metrics to the HANA provider in #915 as well?

**Comment by gdey** [comment]
> @ARolek @gdey
> 
> Can I suggest we merge #916 to fix the viewer in CI, then #893 to add the HANA provider and then #915 after I added unique metrics to the HANA provider in #915 as well?

I'm good with this. @ARolek can you get this reviewed? The changes seem straight forward, but I'm not well verst in Node.

@iwpnd Looks like there is still there the CI is still failing in building the UI?

**Comment by ARolek** [comment]
> Can I suggest we merge https://github.com/go-spatial/tegola/pull/916 to fix the viewer in CI, then https://github.com/go-spatial/tegola/pull/893 to add the HANA provider and then https://github.com/go-spatial/tegola/pull/915 after I added unique metrics to the HANA provider in https://github.com/go-spatial/tegola/pull/915 as well?

Yes! I just merged in #916. We will land HANA provider next.

File: provider/postgis/postgis.go [hunk 2]
```diff
@@ -34,6 +34,11 @@ const Name = "postgis"
 
 type connectionPoolCollector struct {
 	*pgxpool.Pool
+
+	// provider the pool is created for
```
Comment (ARolek) [inline provider/postgis/postgis.go:38]: nit, but the `provider` -> `providerName` in the comment

**Review by ARolek — APPROVED** [review]
Wow! That's way cleaner. Thanks for sticking with it!

**Comment by gdey** [comment]
@iwpnd #893 is merged into mainline, when you have a chance, could you rebase, and update the Hana providers as needed?

File: provider/hana/hana.go [hunk 3]
```diff
@@ -387,7 +394,13 @@ func CreateProvider(config dict.Dicter, maps []provider.Map, providerType string
...
+	name, err := config.String(ConfigKeyName, nil)
+	if err != nil {
+		return nil, err
+	}
+
 	p := Provider{
+		name:      name,
 		dbVersion: uint(majorVersion),
 		srid:      uint64(srid),
 		pool:      &connectionPoolCollector{pool: conn},
```
Comment (gdey) [inline provider/hana/hana.go:406]: Shouldn't you be passing in the ProviderName to the connectionPoolCollector here?

**Review by gdey — CHANGES_REQUESTED** [review]
I think you are forgetting to pass in the provider name to the connection pool for the Hana driver. @mrylov Can you give this a quick review as well? Thank you.

File: provider/hana/hana.go (hunk 3 above)
Comment (iwpnd) [reply provider/hana/hana.go]: good catch, long week 👍

**Review by gdey — APPROVED** [review]
LGTM

**Comment by mrylov** [comment]
Thank you @ARolek, @gdey, @iwpnd for the review!

**Pull request merged**